population/blockgroup2020.py

```diff
@@ -0,0 +1,76 @@
+from common.constants import entity_key
+from entity.abstract import ResourceEntity
+
+import json
+import requests
+
+API_URL = 'https://data.census.gov/api/access/data/table' + \
+    '?id=ACSDT5Y2020.B01003&g=040XX00US{}$1500000'
+
+class BlockGroupPopulation2020(ResourceEntity):
+
+    @staticmethod
+    def dependencies():
+        return[
+            entity_key.census_block_group
+        ]
+
+    def format_block_group(self, block_group):
+        return block_group.replace('ñ','n').lower()
+
+    def get_block_group_id(self, record, field):
+        block_group_entity = self.dependencies_map[entity_key.census_block_group]
+        block_group = block_group_entity.get_cached_value(self.format_block_group(record[field]))
+        return block_group['id'] if block_group else None
+
+    def __init__(self):
+        super().__init__()
+
+        self.table_name = 'block_group_population_2020'
+        self.fields = [
+            {'field': 'population'},
+            {'field': 'block_group', 'column': 'block_group_id', 'data': self.get_block_group_id},
+        ]
+        self.cacheable_fields = ['block_group_id']
+
+    def skip_record(self, record):
+        return self.record_cache and self.format_block_group(record['block_group']) in self.record_cache
+
+    def load_cache(self):
+        if self.record_cache is None:
+            self.record_cache = {}
+
+        if self.cacheable_fields is not None:
+            records = self.mysql_client.select(self.table_name)
+            block_group_entity = self.dependencies_map[entity_key.census_block_group]
+            
+            for record in records:
+                for field in self.cacheable_fields:
+                    block_group_record = block_group_entity.get_cached_value(record[field])
+                    formatted_block_group = block_group_entity.format_block_group(block_group_record['name'])
+                    self.record_cache[formatted_block_group] = record
+
+    def fetch(self):
```

Comment:
This seems like the same function outside of the URL and geographic attribute. Is it possible to create a reusable function that takes in the API_URL and attribute (ex: `block_group`) as parameters and reuse it for block_group, census_tract and zipcode?